pkg/tcpip/transport/tcp/connect.go

```diff
@@ -897,6 +901,10 @@ func (e *endpoint) protocolMainLoop(handshake bool) *tcpip.Error {
 			e.lastErrorMu.Unlock()
 
 			e.mu.Lock()
+			if e.state == StateEstablished || e.state == StateCloseWait {
+				e.stack.Stats().TCP.EstablishedResets.Increment()
+				e.stack.Stats().TCP.CurrentEstablished.Decrement()
+			}
```

Comment:
This is not required. If the handshake failed the connection never became established and hence cannot move to CloseWait.

Reply:
Thanks for the analysis. I'll drop this snippet of code.

Reply:
Not resolved yet?

Reply:
This still needs fixing?